### ISSUE TITLE
Whitespaces are not trimmed in Text Node with `preserveWhitespaces: false`

### ISSUE THREAD

**Issue by JounQin** [angular/angular#37635]
# 🐞 bug report

### Affected Package

The issue is caused by package `@angular/compiler`

### Is this a regression?

 Yes, the previous version in which this bug was not present was: ....

No idea

### Description

 A clear and concise description of the problem...

https://github.com/angular/angular/blob/d1ea1f4c7f3358b730b0d94e65b00bc28cae279c/packages/compiler/src/ml_parser/html_whitespaces.ts#L78

```html
<!-- input -->
<span>
  {{ 'text' }}
</span>

<!-- output -->
<span> {{ 'text' }} </span>


<!-- expected -->
<span>{{ 'text' }}</span>
```

There is no way to `trim` the text.

## 🔬 Minimal Reproduction

 https://stackblitz.com/edit/angular-ivy-pkkssb

## 🌍  Your Environment

**Angular Version:**
<pre><code>

Angular CLI: 9.1.1
Node: 12.18.0
OS: darwin x64

Angular: 9.1.2
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.901.1
@angular-devkit/build-angular     0.901.1
@angular-devkit/build-optimizer   0.901.1
@angular-devkit/build-webpack     0.901.1
@angular-devkit/core              9.1.1
@angular-devkit/schematics        9.1.1
@angular/cdk                      9.2.1
@angular/cli                      9.1.1
@angular/flex-layout              9.0.0-beta.29
@ngtools/webpack                  9.1.1
@schematics/angular               9.1.1
@schematics/update                0.901.1
rxjs                              6.5.5
typescript                        3.8.3
webpack                           4.42.0
</code></pre>

**Anything else relevant?**

**Comment by JoostK** [comment]
This is on purpose, actually. Whitespace is significant in HTML but collapsed to a single whitespace Because of this, the following is equivalent when rendered:

```html
<strong>Bold</strong> text
```
```html
<strong>Bold</strong>        text
```
```html
<strong>Bold</strong>
text
```

However, the following would not be:

```html
<strong>Bold</strong>text
```

as it would join the two words together.

**Comment by pkozlowski-opensource** [comment]
Yep, this is working as designed and we've got tests to prove it: https://github.com/angular/angular/blob/1197965e69d658c92422d49b1f2a1ba90498af67/packages/compiler/test/ml_parser/html_whitespaces_spec.ts#L62-L65

I can see how one would argue for the behaviour specified by @JounQin though.
At the same time @JoostK point is super-valid here (although we've got the described problem already with `<span>two</span> <span>words</span> `

**Comment by petebacondarwin** [comment]
Also this approach is pretty standard across the industry. For example paste the example given into http://minifycode.com/html-minifier or https://htmlcompressor.com/compressor/

**Comment by JounQin** [comment]
@petebacondarwin How about [html-minifier-terser](https://github.com/DanielRuf/html-minifier-terser) https://danielruf.github.io/html-minifier-terser which is used by [html-loader](https://github.com/webpack-contrib/html-loader)?

![image](https://user-images.githubusercontent.com/8336744/85003558-f3ab9b80-b188-11ea-8310-f9993499e3f3.png)

**Comment by petebacondarwin** [comment]
Two things to note with that tool:

1. It has a "conservative collapse" option which is the same as our collapsing strategy.
2. It tries to be intelligent in cases where there might be a problem:

<img width="441" alt="Screenshot 2020-06-18 at 10 39 40" src="https://user-images.githubusercontent.com/15655/85005160-52a0ee80-b150-11ea-8d9c-d8873cf9aa55.png">

As @pkozlowski-opensource mentioned, there is room for discussion here but I don't think this would be a high priority for us.

**Comment by dgardner-nv** [comment]
Just wrap the interpolation in `ng-container` and it won't apply the white spaces around it.

I would argue that this is more of a problem than stated.  Consider multi-line text that's been wrapped.  Since an unnecessary white space is added before the interpolated text, it appears that the text is indented by one space on the first line.  It also creates issues around spacing because you have to account for an extra white space before and after the interpolated text.  If I want a space, I'll add a space.  Don't need it added automatically for me.

**Comment by JoostK** [comment]
> Just wrap the interpolation in `ng-container` and it won't apply the white spaces around it.
> 
> ```
> <div>
>     <ng-container>{{ value }}</ng-container>
> </div>
> ```
> 
> I would argue that this is more of a problem than stated. Consider multi-line text that's been wrapped. Since an unnecessary white space is added before the interpolated text, it appears that the text is indented by one space on the first line. It also creates issues around spacing because you have to account for an extra white space before and after the interpolated text. If I want a space, I'll add a space. Don't need it added automatically for me.

@dgardner-nv Angular does not introduce spaces into templates. It only removes non-significant whitespace. It may be that you're using a formatter that introduces whitespace, but if this affects the rendered output then that's a bug in your formatter.

**Comment by pchabros** [comment]
To overcome this issue you can set [`htmlWhitespaceSensitivity`](https://prettier.io/docs/en/options.html#html-whitespace-sensitivity) to `"strict"` in Prettier config.
Then this:
```html
<span>Very very very very very long text</span>
```
will be formatted into this:
```html
<span
  >Very very very very very long text</span
>
```
instead of this:
```html
<span>
  Very very very very very long text
</span>
```
